### DISCUSSION TITLE
Starship prompt does not change without typing source ~/.bash_profile manually

### DISCUSSION THREAD

**Discussion by noraworld** [starship/starship#1545] [Q&A] [answered]
In my environment, the Starship prompt does not change if I don’t type `source ~/.bash_profile` manually.

Of course, `~/.bash_profile` is loaded when new terminal is open. `~/.bash_profile` loads `~/.bashrc`, and `eval "$(starship init bash)"` is written in `~/.bashrc`.

But the Starship prompt does not change just opening the terminal. I type `source ~/.bash_profile` manually, and then it works.

Please tell me how to deal with this problem.

**Comment by noraworld** [comment]
Here’s my environment.

* [Starship](https://github.com/noraworld/dotfiles/blob/ca68a5f43664d6f2215c1006597d669fd80be558/Brewfile.lock.json#L630-L650)
* [System](https://github.com/noraworld/dotfiles/blob/ca68a5f43664d6f2215c1006597d669fd80be558/Brewfile.lock.json#L722-L733)
* [.bashrc](https://github.com/noraworld/dotfiles/blob/ca68a5f43664d6f2215c1006597d669fd80be558/bashrc#L220)
* [.bash_profile](https://github.com/noraworld/dotfiles/blob/ca68a5f43664d6f2215c1006597d669fd80be558/bash_profile)

**Comment by noraworld** [comment, marked as answer]
Resolved by putting `eval "$(starship init bash)"` above `bash-preexec.sh`.

**Reply by chipbuster** [reply]
Is the bash-preexec script the one [from rcaloras](https://github.com/rcaloras/bash-preexec)? If so, that's probably a bug in our init, since we're supposed to explicitly support that framework.

**Reply by chipbuster** [reply]
Nevermind. From their README: "It (bash-preexec.sh) must be the last thing imported in your bash profile."

We probably do need to document exactly how starship interacts with that script a little better though.
